### PR TITLE
perf: Reduce loops in when clean queue in BroadcastHub

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation:
refs: https://github.com/apache/pekko/issues/1627

Modification:
Reduce the loop count

Result:
Reduce the loop count

@andreaslochbihler-da cc as he is the origin reporter

### CODE REVIEW

**Comment by He-Pin** [comment]
it's better not to change the code in `unblockIfPossible` to 
```scala
        var upToOffset = head
        while (consumerWheel(upToOffset & WheelMask).isEmpty) {
          upToOffset += 1
          unblocked = true
        }
        if (upToOffset != head) {
          cleanQueueInRange(head, upToOffset)
          head = upToOffset
        }
```

**Comment by pjfanning** [comment]
Can we keep this till v1.2.0? We've had a couple of breaks due to optimisations like this.

**Comment by He-Pin** [comment]
I'm not sure if this should be marked as a bug or not.

File: stream/src/main/scala/org/apache/pekko/stream/scaladsl/Hub.scala [hunk 1]
```diff
@@ -605,10 +605,7 @@ private[pekko] class BroadcastHub[T](startAfterNrOfConsumers: Int, bufferSize: I
             else if (head != finalOffset) {
               // If our final consumer goes away, we roll forward the buffer so a subsequent consumer does not
               // see the already consumed elements. This feature is quite handy.
-              while (head != finalOffset) {
-                queue(head & Mask) = null
-                head += 1
-              }
```
Comment (He-Pin) [inline stream/src/main/scala/org/apache/pekko/stream/scaladsl/Hub.scala:611]: Which can loop many times

**Review by pjfanning — APPROVED** [review]
lgtm

**Comment by He-Pin** [comment]
@pjfanning I want to backport to 1.1.x too, wdyt

**Pull request merged**